examples/proxy-server/src/main/java/example/armeria/proxy/ProxyService.java

```diff
@@ -0,0 +1,96 @@
+import com.linecorp.armeria.common.FilteredHttpResponse;
+import com.linecorp.armeria.common.HttpHeaderNames;
+import com.linecorp.armeria.common.HttpHeaders;
+import com.linecorp.armeria.common.HttpObject;
+import com.linecorp.armeria.common.HttpRequest;
+import com.linecorp.armeria.common.HttpResponse;
+import com.linecorp.armeria.common.SessionProtocol;
+import com.linecorp.armeria.server.AbstractHttpService;
+import com.linecorp.armeria.server.ServiceRequestContext;
+
+public final class ProxyService extends AbstractHttpService {
+
+    /**
+     * We used hardcoded backend address. But we can use service discovery mechanism to configure backends
+     * dynamically using {@link DnsServiceEndpointGroup}, ZooKeeper or Central Dogma.
+     *
+     * See <a href="https://line.github.io/armeria/client-service-discovery.html">service discovery</a>,
+     * <a href="https://line.github.io/armeria/advanced-zookeeper.html#advanced-zookeeper">Service discovery
+     * with ZooKeeper</a> and <a href="https://line.github.io/centraldogma">Central Dogma</a>.
+     */
+    private static final EndpointGroup animationGroup = new StaticEndpointGroup(
+            Endpoint.of("127.0.0.1", 8081),
+            Endpoint.of("127.0.0.1", 8082),
+            Endpoint.of("127.0.0.1", 8083));
+
+    private final HttpClient loadBalancingClient;
+
+    private final boolean filterHeaders;
+
+    public ProxyService() throws InterruptedException {
+        loadBalancingClient = newLoadBalancingClient();
+        filterHeaders = false;
+    }
+
+    private static HttpClient newLoadBalancingClient() throws InterruptedException {
+        // Send HTTP health check requests to '/internal/l7check' every 10 seconds.
+        final HttpHealthCheckedEndpointGroup healthCheckedGroup =
+                new HttpHealthCheckedEndpointGroupBuilder(animationGroup, "/internal/l7check")
+                        .protocol(SessionProtocol.HTTP)
+                        .retryInterval(Duration.ofSeconds(10))
+                        .build();
+
+        // Wait until the initial health check is finished.
+        healthCheckedGroup.awaitInitialEndpoints();
+
+        EndpointGroupRegistry.register("animation_apis", healthCheckedGroup,
+                                       // You can use EndpointSelectionStrategy.WEIGHTED_ROUND_ROBIN or even
+                                       // implement your own strategy to balance requests.
+                                       EndpointSelectionStrategy.ROUND_ROBIN);
+
+        return new HttpClientBuilder("http://group:animation_apis")
+                // Increase timeout to serve long streaming response.
+                .defaultResponseTimeout(Duration.ofHours(1))
+                .build();
+    }
+
+    @Override
+    protected HttpResponse doGet(ServiceRequestContext ctx, HttpRequest req) throws Exception {
+        final HttpResponse res = loadBalancingClient.execute(req);
```

Comment:
How about adding some comment for this line? e.g. What it does, why no backpressure control is here, etc.

Reply:
Added. Not sure it explains well though. 😄